src/main/java/org/minions/devfund/mauricioz/Battleship/Ship.java

```diff
@@ -0,0 +1,209 @@
+
+    /**
+     * Abstract method to get the ship type.
+     * @return shipType.
+     */
+    public abstract String getShipType();
+
+    /**
+     * Method to review if it is possible place a ship on a given coordinates.
+     * @param row coordinate.
+     * @param column coordinate.
+     * @param horizontal position of the ship.
+     * @param ocean object.
+     * @return true or false.
+     */
+    public boolean okToPlaceShipAt(int row, int column, boolean horizontal, final Ocean ocean) {
+        if (startOrFinishOnTheEdge(row, column, ocean) || nothingAround(row, column, horizontal, ocean)) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * verify is not another ship near of the desired position.
+     * @param row value.
+     * @param column value.
+     * @param horizontal ship direction.
+     * @param ocean board.
+     * @return true or false.
+     */
+    private boolean nothingAround(int row, int column, boolean horizontal, final Ocean ocean) {
+        if (horizontal && nothingAroundForPlacedHorizontal(row, column, ocean)) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * verify is no another ship near of the desired position to place the ship.
+     * @param row value.
+     * @param column value.
+     * @param ocean board.
+     * @return true or false.
+     */
+    private boolean nothingAroundForPlacedHorizontal(int row, int column, final Ocean ocean) {
+        boolean isEmpty = false;
+        for (int i = 0; i < row; i++) {
+            for (int j = 0; j < column; j++) {
+                if (ocean.getShips()[i][j].getShipType() == "empty") {
+                    isEmpty = true;
+                } else {
+                    isEmpty = false;
+                }
+            }
+        }
+        return isEmpty;
+    }
+
+    /**
+     * validate ship position is on the edge of the board.
+     * @param row value.
+     * @param column value.
+     * @param ocean board.
+     * @return true or false.
+     */
+    private boolean startOrFinishOnTheEdge(int row, int column, final Ocean ocean) {
+        return row == 0 || column == 0
+                || row + this.getLength() == ocean.getShips().length - 1
+                || column + this.getLength() == ocean.getShips().length - 1;
+    }
+
+    /**
+     * action of place a ship onto a valid space into the ocean.
+     * @param row coordinate.
+     * @param column coordinate.
+     * @param horizontal orientation of the ship.
+     * @param ocean object.
+     */
+    public void placeShipAt(int row, int column, boolean horizontal, final Ocean ocean) {
+        //TODO
+    }
+
+    /**
+     * If a part of the ship occupies the given row and column, and the ship hasn’t been sunk,
+     * mark that part of the ship as ”hit”.
+     * @param row coordinate.
+     * @param column coordinate.
+     * @return true or false
+     */
+    public boolean shootAt(int row, int column) {
+        //TODO
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)